components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -960,6 +960,8 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
           lamr(i,k),mu_r(i,k),cdistr(i,k),qr_incld(i,k),&
                qrheti,nrheti)
 
+          ! AaronDonahue: The following should be removed or moved to a separate
+          ! subroutine? 
```

Comment:
I like having a reminder that rime splintering is a thing in the code. It shouldn't be in a separate subroutine since we want the top-level to remind us of the processes one might include. I thought I put a comment in here saying rime splintering is omitted when only 1 ice category is used to compensate for the fact that ice is being forced to fit a single distribution? I guess I'm saying that we should leave the comment as is.

Reply:
Does that extend to the calculation of aaa and xx?  In the original code (which will be changed) both aaa and xx are used for the rain/liq calculations and for the ice calculations so I decided to keep them separate from the individual subroutines.

Reply:
I'm confused why you bring up aaa and xx. I was just saying we should keep the ice multiplication header comment and my subsequent comment intact. Are you envisioning deleting a much larger swath of code? If so, can you somehow indicate the start and end points of the code you want to remove? Or we could talk about it in person...

Reply:
i only brought it up because the full effort is to compartmentalize the code in such a way that we have a set of individual subroutines and this is a big chunk of code. We can’t remove it, i’m just wondering if we want to make a separate subroutine for calculating xx and aaa to keep this part of the code clean.

Reply:
I've moved the calculation of xx and aaa to it's own routine.